### PR TITLE
Build lookup improvements

### PR DESCRIPTION
**Pull request by mmitche**
- Add new optional parameters to lookup builds by the AzDO information
- Add new darc commands to lookup builds by repo+commit or the azdo URI.
  The azdo URI option is currently disabled as it requires the new REST api.

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs [hunk 1]
```diff
@@ -33,22 +34,79 @@ public override async Task<int> ExecuteAsync()
...
                 }
                 else
                 {
-                    Console.WriteLine($"Could not find build with id '{_options.Id}'");
+                    Console.WriteLine("Please specify --id, --uri, or --repo and --commit to lookup a build.");
                     return Constants.ErrorCode;
                 }
 
+                // Print the build info.
+                if (!matchingBuilds.Any())
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs:83]: This is technically a possible multiple enumeration of this enum (which can do weird things for some flavors of enumerations). They are almost certainly just Lists or Arrays under the hoods, so it probably doesn't matter.

File: src/Microsoft.DotNet.Darc/src/Darc/Options/GetBuildCommandLineOptions.cs [hunk 2]
```diff
@@ -10,9 +10,18 @@ namespace Microsoft.DotNet.Darc.Options
...
+        [Option("id", HelpText = "Build id.")]
         public int Id { get; set; }
 
+        [Option("uri", HelpText = "Uri of the build.")]
+        public string BuildUri { get; set; }
+
+        [Option("repo", HelpText = "Repository")]
+        public string Repo { get; set; }
+
+        [Option("commit", HelpText = "Commit")]
```
Comment (ChadNedzlek) [inline src/Microsoft.DotNet.Darc/src/Darc/Options/GetBuildCommandLineOptions.cs:22]: "commit" isn't a super descriptive help text.

**Comment by mmitche** [comment]
@ChadNedzlek Forgive my forever forgetting of the rules on REST APIs..but adding optional parameters to existing APIs is acceptable, correct?

**Comment by ChadNedzlek** [comment]
@mmitche Yup, as long as the old request would do the same, right thing, which should be the case with an optional parameter: an old person that didn't know about it, so the behavior when it's missing should be the "old" behavior.

Then it's fine. :-)

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs [hunk 3]
```diff
@@ -33,22 +34,79 @@ public override async Task<int> ExecuteAsync()
...
+                {
+                    Console.WriteLine($"Could not any builds matching the given criteria.");
```
Comment (JohnTortugo) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GetBuildOperation.cs:85]: NIT: typo, missing "find".

**Pull request merged**